packages/gestalt/src/AvatarGroup.js

```diff
@@ -0,0 +1,265 @@
+// @flow strict
+import React, { useState, type Node } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import Avatar from './Avatar.js';
+import Box from './Box.js';
+import Flex from './Flex.js';
+import Text from './Text.js';
+import Icon from './Icon.js';
+import TapArea from './TapArea.js';
+import { CompositeZIndex, FixedZIndex } from './zIndex.js';
+import { useColorScheme } from './contexts/ColorScheme.js';
+import { type AbstractEventHandler } from './AbstractEventHandler.js';
+import styles from './AvatarGroup.css';
+
+type Size = 'xs' | 'sm' | 'md';
+
+type Role = 'link' | 'button' | null;
+
+type Collaborators = $ReadOnlyArray<{|
+  name: string,
+  src?: string,
+|}>;
+
+type Props = {|
+  accessibilityLabel: string,
+  accessibilityControls?: string,
+  accessibilityExpanded?: boolean,
+  accessibilityHaspopup?: boolean,
+  addCollaborators?: boolean,
+  collaborators: Collaborators,
+  href?: string,
+  onClick?: AbstractEventHandler<
+    | SyntheticMouseEvent<HTMLDivElement>
+    | SyntheticKeyboardEvent<HTMLDivElement>
+    | SyntheticMouseEvent<HTMLAnchorElement>
+    | SyntheticKeyboardEvent<HTMLAnchorElement>,
+    {| disableOnNavigation: () => void |},
+  >,
+  role?: Role,
+  size?: Size,
+|};
+
+type CollaboratorAvatarType = ({|
+  hovered: boolean,
+  index: number,
+  name: string,
+  size: Size,
+  src: string,
+|}) => Node;
+
+type AddCollaboratorsBtnType = ({| hovered: boolean |}) => Node;
+
+type CollaboratorsCountType = ({|
+  hasAddCollaboratorsBtn: boolean,
+  hovered: boolean,
+  count: number,
+|}) => Node;
+
+type HoverOverlayType = ({|
+  children: Node,
+  hovered: boolean,
+  size: Size,
+|}) => Node;
+
+const BASE_ZINDEX = new FixedZIndex(1); // hover overlay needs z-index = 1
+
+const COLLABORATOR_COUNT_ZINDEX = new FixedZIndex(2); // hover overlay needs z-index = 1
```

Comment:
This comment doesn't make sense here…copypasta from the previous line? Also, should this be a CompositeZIndex?

Reply:
OUTDATED